pulp_rpm/app/models.py

```diff
@@ -30,6 +30,141 @@ class RpmContent(Content):
     #     )
 
 
+class UpdateRecord(Content):
+    """
+    The "UpdateRecord" content type. Formerly "Errata".
+
+    Maps directly to the fields provided by createrepo_c
+    https://github.com/rpm-software-management/createrepo_c/
+
+    Fields:
+        id (Text):
+            Update id (short update name, e.g. RHEA-2013:1777)
+        updated_date (Text):
+            Date when the update was updated (e.g. "2013-12-02 00:00:00")
+
+        description (Text):
+            Update description
+        issued_date (Text):
+            Date when the update was issued (e.g. "2013-12-02 00:00:00")
+        fromstr (Text):
+            Source of the update (e.g. security@redhat.com)
+        status (Text):
+            Update status ("final", ...)
+        title (Text):
+            Update name
+        summary (Text):
+            Short summary
+        version (Text):
+            Update version (probably always an integer number)
+
+        type (Text):
+            Update type ("enhancement", "bugfix", ...)
+        severity (Text):
+            Severity
+        solution (Text):
+            Solution
+        release (Text):
+            Update release
+        rights (Text):
+            Copyrights
+
+        pushcount (Text):
+            Push count
+
+        collections (Text):
+            List of UpdateCollections
+        references (Text):
+            List of UpdateReferences
+    """
+
+    # Required metadata
+    id = models.TextField(required=True)
+    updated_date = models.TextField(required=True)
+
+    # Optional metadata
+    description = models.TextField(blank=True)
+    issued_date = models.TextField(blank=True)
+    fromstr = models.TextField(blank=True)  # formerly "errata_from"
+    status = models.TextField(blank=True)
+    title = models.TextField(blank=True)
+    summary = models.TextField(blank=True)
+    version = models.TextField(blank=True)
+
+    type = models.TextField(blank=True)
+    severity = models.TextField(blank=True)
+    solution = models.TextField(blank=True)
+    release = models.TextField(blank=True)
+    rights = models.TextField(blank=True)
+
+    pushcount = models.TextField(blank=True)
+
+    collections = models.TextField(blank=True)  # formerly "pkglist"
+    references = models.TextField(blank=True)
+
```

Comment:
With UpdateCollections and UpdateReferences, much like with "depends" we can either go the route of JSON representation, or make actual models with a foreign key that provides a related_name of  "collections".

Do we have a preference?  Is there value in modelling it vs. shoving it into JSON?

Reply:
Pulp 2 did model "pkglist" / collections separately

Reply:
+1 to JSON for simplicity. Are we using the custom json field or just storing it as a StringField and serializing/deserailizing when we read/write?

Reply:
TextField for now... messing with JSONField directly is fraught with enough pitfalls generally speaking that until further notice I wouldn't want to rely on it.  It works fine for reading (how we use it with Tasks) but writing is really hard to make work properly